### PR TITLE
gdtoolkit: init at 3.3.1

### PR DESCRIPTION
**Pull request by shiryel**
###### Motivation for this change

Add package https://github.com/Scony/godot-gdscript-toolkit

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/gdtoolkit/default.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  src = fetchFromGitHub {
+    owner = "Scony";
+    repo = "godot-gdscript-toolkit";
+    rev = version;
+    sha256 = "13nnpwy550jf5qnm9ixpxl1bwfnhhbiys8vqfd25g3aim4bm3gnn";
+  };
+
+  disabled = pythonOlder "3.7";
+
+  # Tests cannot be run because they need network to install additional dependencies using pip and tox
```
Comment (IvarWithoutBones) [inline pkgs/development/python-modules/gdtoolkit/default.nix:36]: Have you tried to add those dependencies in advance using `checkInputs`?

File: pkgs/development/python-modules/gdtoolkit/default.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  doCheck = false;
+  pythonImportsCheck = [ "gdtoolkit" "gdtoolkit.formatter" "gdtoolkit.linter" "gdtoolkit.parser" ];
+
+  meta = {
```
Comment (IvarWithoutBones) [inline pkgs/development/python-modules/gdtoolkit/default.nix:40]: It looks like `meta.maintainers` is missing, which is required for new packages

File: pkgs/development/python-modules/gdtoolkit/default.nix (hunk 1 above)
Comment (shiryel) [reply pkgs/development/python-modules/gdtoolkit/default.nix:36]: Yes, then they tried to download a lot of other dependencies and failed

File: pkgs/development/python-modules/gdtoolkit/default.nix (hunk 2 above)
Comment (shiryel) [reply pkgs/development/python-modules/gdtoolkit/default.nix]: done :)

**Comment by IvarWithoutBones** [comment]
The code looks good :blush: 

Just a few issues with the commits, could you rename them to something like this?
```bash
* maintainers: add shiryel # first
* gdtoolkit: init at 3.3.1 # second
```
That way you can also include the `meta.maintainers` addition in the original commit, since we want to have every commit building/abiding guidelines.

**Review by IvarWithoutBones — APPROVED** [review]
Looks good to me, thanks! The binary works as well.

<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.gdtoolkit</li>
    <li>python39Packages.gdtoolkit</li>
  </ul>
</details>

File: pkgs/development/python-modules/gdtoolkit/default.nix [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+let lark080 = lark.overrideAttrs (old: rec {
+  # gdtoolkit needs exactly this lark version
+  version = "0.8.0";
+  src = fetchFromGitHub {
+    owner = "lark-parser";
+    repo = "lark";
+    rev = version;
+    sha256 = "su7kToZ05OESwRCMPG6Z+XlFUvbEb3d8DgsTEcPJMg4=";
+  };
+});
```
Comment (mweinelt) [inline pkgs/development/python-modules/gdtoolkit/default.nix:12]: Overrides like that are no allowed when using `buildPythonPackage`.

See https://github.com/NixOS/nixpkgs/blob/master/doc/languages-frameworks/python.section.md#using-pythonrelaxdepshook-using-pythonrelaxdepshook

**Pull request merged**